test/com/amazon/ion/IonTestCase.java

```diff
@@ -828,7 +817,7 @@ public static void checkSymbol(String name, int id, IonValue value)
 
         assertFalse(value.isNullValue());
 
-        if (name == null && id != 0)
```

Comment:
This was wrong. `stringValue()` never should have returned `null` for symbol zero.